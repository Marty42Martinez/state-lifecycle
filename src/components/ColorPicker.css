.ColorPicker {
  background: black;
  width: 15em;
  
  button {
    border: none;
    border-radius: 20%;
    padding: 1em;
    margin: 0.5em;
    cursor: pointer;
  }

  div {
    width: 100px;
    height: 100px;
  }

  .red {
    background-color: red;
  }

  .darkorchid {
    background-color: darkorchid;
    color: white;
  }

  .blue {
    background-color: blue;
    color: white;
  }

}