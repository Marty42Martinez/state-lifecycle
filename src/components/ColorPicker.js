import React, { PureComponent } from 'react';
//import styles

export default class ColorPicker extends PureComponent {
  state = {
    color: ''
  };

  selectColor = color => {
    this.setState({ color });
  }

  render() {
    const { color } = this.state;

    return (
      <section>
        <button onClick={this.selectColor.bind(null, 'red')}>Red</button>
        <button onClick={() => this.selectColor('yellow')}>Yellow</button>
        <button onClick={() => this.selectColor('blue')}>Blue</button>
        <div></div>
      </section>
    );
  }
}

