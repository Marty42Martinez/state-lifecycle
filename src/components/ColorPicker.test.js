import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker';

describe('Color Picker component', () => {
  it('can create an instance of Color Picker', () => {
    const wrapper = shallow(<ColorPicker selectColorHandler={jest.fn()}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('selects red on button click', () => {
    const selectColorHandler = jest.fn();
    const wrapper = shallow(<ColorPicker selectColorHandler={selectColorHandler} />);
    wrapper.find('button').at(0).simulate('click');

    expect(selectColorHandler).toHaveBeenCalledWith('red');
  });

});
