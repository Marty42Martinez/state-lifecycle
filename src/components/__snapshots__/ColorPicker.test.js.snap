// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker component renders a ColorPicker 1`] = `
<section
  className="ColorPicker"
>
  <button
    className="red"
    onClick={[Function]}
  >
    Red
  </button>
  <button
    className="darkorchid"
    onClick={[Function]}
  >
    Dark Orchid
  </button>
  <button
    className="blue"
    onClick={[Function]}
  >
    Blue
  </button>
  <div
    className=""
  />
</section>
`;
